src/blockdev.rs

```diff
@@ -300,7 +300,16 @@ impl PartTableKpartx {
 
 impl PartTable for PartTableKpartx {
     fn reread(&mut self) -> Result<()> {
-        self.run_kpartx("-u")
+        let delay = 1;
+        for _ in 0..5 {
+            match self.run_kpartx("-u") {
+                Ok(()) => return Ok(()),
+                Err(e) => eprintln!("Error: {}", e),
+            }
+            eprintln!("Retrying in {} second", delay);
+            sleep(Duration::from_secs(delay));
```

Comment:
Minor: this will print and sleep one more time on the last iteration of the loop even though we won't really be retrying.

One approach I sometimes use in that pattern is to just retry one last time outside the loop since we'll fail anyway. That way you can also return that error itself. I.e. ...